plugins/linux/l3plugin/descriptor/route.go

```diff
@@ -122,10 +122,15 @@ func NewRouteDescriptor(
 // EquivalentRoutes is case-insensitive comparison function for l3.LinuxRoute.
 func (d *RouteDescriptor) EquivalentRoutes(key string, oldRoute, newRoute *linux_l3.Route) bool {
 	// attributes compared as usually:
-	if oldRoute.OutgoingInterface != newRoute.OutgoingInterface ||
-		oldRoute.Scope != newRoute.Scope {
+	if oldRoute.OutgoingInterface != newRoute.OutgoingInterface {
 		return false
 	}
+	// compare scopes for IPv4 routes
+	if ip := net.ParseIP(strings.Split(newRoute.DstNetwork, "/")[0]); ip != nil && ip.To4() != nil {
```

Comment:
Errrm.. I am not sure if my [previous comment](https://github.com/ligato/vpp-agent/pull/1829#discussion_r782056354) prompted you to remove the helper function.. if so then it was not my intention. I was commenting on the function being a method. I think it's better to provide little helper function for this code, as it is repeated several times.

Reply:
I'm reading the previous comment again and it still sounds to me that you don't like the helper function itself 🙂 which - without the logging - is basically a one-liner anyway. Ok no matter, let's agree that I'll keep it there as an unexported function.

Reply:
Apologies for my unlucky wording.